File: Houseclub/src/main/java/me/grishka/houseclub/api/methods/GetChannel.java
package me.grishka.houseclub.api.methods;

import me.grishka.houseclub.api.ClubhouseAPIRequest;
import me.grishka.houseclub.api.model.Channel;

public class GetChannel extends ClubhouseAPIRequest<Channel> {
    public GetChannel(String name) {
        super("POST", "get_channel", Channel.class);
        requestBody = new Body(name);
    }

    private static class Body {
        public String channel;

        public Body(String channel) {
            this.channel = channel;
        }
    }
}

File: Houseclub/src/main/java/me/grishka/houseclub/api/methods/GetChannel.kt
package me.grishka.houseclub.api.methods

import me.grishka.houseclub.api.ClubhouseAPIRequest
import me.grishka.houseclub.api.model.Channel

class GetChannel(name: String) : ClubhouseAPIRequest<Channel?>("POST", "get_channel", Channel::class.java) {
    private class Body(var channel: String)

    init {
        requestBody = Body(name)
    }
}